File: backend/src/models/SmsLog.js
'use strict';

const { Model, DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  class SmsLog extends Model {
    static associate(models) {
      SmsLog.belongsTo(models.Member, { foreignKey: 'sender_id', as: 'sender' });
    }
  }

  SmsLog.init({
    id: { type: DataTypes.BIGINT, primaryKey: true, autoIncrement: true, allowNull: false },
    sender_id: { type: DataTypes.BIGINT, allowNull: false },
    role: { type: DataTypes.STRING(50), allowNull: false },
    recipient_type: { type: DataTypes.ENUM('individual', 'group', 'all'), allowNull: false },
    recipient_member_id: { type: DataTypes.BIGINT, allowNull: true },
    group_id: { type: DataTypes.BIGINT, allowNull: true },
    recipient_count: { type: DataTypes.INTEGER, allowNull: false, defaultValue: 1 },
    message: { type: DataTypes.TEXT, allowNull: false },
    status: { type: DataTypes.ENUM('success', 'partial', 'failed'), allowNull: false },
    error: { type: DataTypes.TEXT, allowNull: true }
  }, {
    sequelize,
    modelName: 'SmsLog',
    tableName: 'sms_logs',
    timestamps: true,
    underscored: true
  });

  return SmsLog;
};

File: backendJava/src/main/java/church/abunearegawi/backend/model/SmsLog.java
package church.abunearegawi.backend.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "sms_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SmsLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "sender_id", nullable = false)
    private Member sender;

    @Column(nullable = false, length = 50)
    private String role;

    @Enumerated(EnumType.STRING)
    @Column(name = "recipient_type", nullable = false)
    private RecipientType recipientType;

    @Column(name = "recipient_member_id")
    private Long recipientMemberId;

    @Column(name = "group_id")
    private Long groupId;

    @Column(name = "department_id")
    private Long departmentId;

    @Column(name = "recipient_count", nullable = false)
    @Builder.Default
    private Integer recipientCount = 1;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(columnDefinition = "TEXT")
    private String error;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum RecipientType {
        INDIVIDUAL("individual"),
        GROUP("group"),
        DEPARTMENT("department"),
        ALL("all");

        private final String value;

        RecipientType(String value) {
            this.value = value;
        }
    }

    public enum Status {
        SUCCESS("success"),
        PARTIAL("partial"),
        FAILED("failed");

        private final String value;

        Status(String value) {
            this.value = value;
        }
    }
}
